### ISSUE TITLE
[CLOSED] Use metaprogramming to cast integers in sf::Packet

### ISSUE THREAD

**Issue by RobinPetit** [WizardCorp/WizardPoker#88]
<a href="https://github.com/TheoVerhelst"><img src="https://avatars.githubusercontent.com/u/9685883?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [TheoVerhelst](https://github.com/TheoVerhelst)**
_Wednesday Mar 09, 2016 at 09:25 GMT_
_Originally opened as https://github.com/ulb-infof209/Group5/pull/88_

---

With this method, the type that should be used in a packet to transmit an integer type is automatically found by using some meta-programming tricks.

Yes I should focus on making the game working before doing things like this, but damn, this is FUN!

These commits add many dirty template structs, but once this is done, this is easier to add and extract integers from packets, there will be an unique instruction to do this, no matter of the type of the integer, even if this is a typedef (such as the type `std::vector<>::size_type` returned by `std::vector::size`).

Obviously I'll not leave the template structs in the class Game, if some people approve my craziness, I'll make a dedicated file and replacing all integers casts already done.

---

_**[TheoVerhelst](https://github.com/TheoVerhelst)** included the following code: https://github.com/ulb-infof209/Group5/pull/88/commits_
